### PR TITLE
Don't fall back to `help` command when command is empty

### LINKED ISSUE #24577: Bazel does not fail for empty command
### Description of the bug:

When running `bazel '' test ...` (note the empty argument) Bazel just displays its default help text (as if just running `bazel`) and exits with code `0`.

When using an invalid command like `bazel doflips ...` it correctly reports an error and exits with code `2`.

Context: I got bitten by this when the empty argument sneaked into the command line, CI was green when it actually wasn't.

### Which category does this issue belong to?

CLI

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run `bazel '' test ...`

### Which operating system are you running Bazel on?

Windows, Linux

### What is the output of `bazel info release`?

5.4.1 - 8.0.0rc7

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

_No response_

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?

_No response_

### If this is a regression, please try to identify the Bazel commit where the bug was introduced with bazelisk --bisect.

_No response_

### Have you found anything relevant by searching the web?

_No response_

### Any other information, logs, or outputs that you want to share?

_No response_

### PR DESCRIPTION
**Pull request by fmeum**
This masked issues in Bazel wrappers in a very confusing way.

Fixes #24577

### CODE REVIEW

**Comment by lberki** [comment]
How about emitting a user-friendly error message ("No command specified. Call `bazel help` for instructions or see http://bazel/build for extensive documentation?") and returning with a non-zero exit code?

Either way, please add a test case.

**Comment by fmeum** [comment]
@lberki Done. Note that this about the command being the empty string, not about there being no command (which is valid and should forward to `bazel help`).

File: src/main/cpp/option_processor.cc [hunk 1]
```diff
@@ -151,6 +151,11 @@ std::unique_ptr<CommandLine> OptionProcessor::SplitCommandLine(
         std::move(path_to_binary), std::move(startup_args), "", {}));
   }
   string& command = args[i];
+  // Distinguish an empty command from the case of no command above.
+  if (command.empty()) {
+    blaze_util::StringPrintf(error, "Command cannot be the empty string (try 'help')");
```
Comment (lberki) [inline src/main/cpp/option_processor.cc:156]: nit: mind making this symmetric with the "no command" case and say "try 'bazel help'" instead?

File: src/main/cpp/option_processor.cc [hunk 2]
```diff
@@ -151,6 +151,11 @@ std::unique_ptr<CommandLine> OptionProcessor::SplitCommandLine(
...
+    return nullptr;
```
[the remark this answers is not in the document]
Comment (fmeum) [reply src/main/cpp/option_processor.cc:161]: It already is in the case of `bazel ''`, as demonstrated by the test. Which other case are you referring to? Just `bazel` probably shouldn't fail as users may rely on it as a simple "does Bazel work" check.

[the remark this answers is not in the document]
Comment (lberki) [reply src/main/cpp/option_processor.cc:161]: Nevermind, some wires crossed in my brain.

**Review by lberki — APPROVED** [review]

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by iancha1992** [comment]
@bazel-io fork 8.1.0

**Pull request closed without merging**